src/MediaCollections/Filesystem.php

```diff
@@ -129,26 +131,38 @@ public function copyToMediaLibrary(string $pathToFile, Media $media, ?string $ty
             : $media->getDiskDriverName();
 
         if ($diskDriverName === 'local') {
-            $this->filesystem
+            $success = $this->filesystem
                 ->disk($diskName)
                 ->put($destination, $file);
 
             fclose($file);
 
+            if (!$success) {
+                Throw DiskCannotBeAccessed::create($diskName);
+
+                DB::rollback();
```

Comment:
Not sure if the exception or rollback should come first.

Reply:
I refactored and removed this; see comment below.